crates/air-lib/interpreter-data/src/interpreter_data.rs

```diff
@@ -32,16 +32,40 @@ use air_utils::measure;
 use serde::Deserialize;
 use serde::Serialize;
 
-/// The AIR interpreter could be considered as a function
-/// f(prev_data: InterpreterData, current_data: InterpreterData, ... ) -> (result_data: InterpreterData, ...).
-/// This function receives prev and current data and produces a result data. All these data
-/// have the following format.
+#[derive(Debug, thiserror::Error)]
+pub enum DataDeserializationError {
+    #[error("failed to deserialize envelope: {0}")]
+    Envelope(<InterpreterDataEnvRepr as Representation>::DeserializeError),
+    #[error("failed to deserialize data: {0}")]
+    Data(crate::rkyv::RkyvDeserializeError),
+}
+
+/// An envelope for the AIR interpreter data that makes AIR data version info accessible in a stable way.
 #[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct InterpreterData {
+pub struct InterpreterDataEnv {
```

Comment:
Suggest name it `InterpreterDataEnvelope`, otherwise it could be read as `...Environment`.

Reply:
Also I'd say that it's better to distinguish in a clear way that there is a partially parced envelope and fully parsed `InterpreterData`, so I suggest for `try_from_slice` return `Self`, then check version on the preparation step, and only then parse the remainder.

It will make code clearer (without pair of `Versions` and `InterpreterData`) and the fail path without parsing in this case unnecessary `InterpreterData`.